File: tp-arithmetic-java/src/main/java/fr/isima/filters/CountTypeOfOperationFilter.java
package fr.isima.filters;

import fr.isima.service.OperationCounter;

import java.io.IOException;

import javax.servlet.Filter;
import javax.servlet.FilterChain;
import javax.servlet.FilterConfig;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.http.HttpServletRequest;


/**
 * Compte les requêtes en fonction de leur type (math ou chaine)
 * 
 * @author Benjamin Kuchcik
 *
 */
public class CountTypeOfOperationFilter implements Filter {
    private OperationCounter operationCounter;

    @Override
    public void destroy() {

    }

    @Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
        final HttpServletRequest httpReq = (HttpServletRequest) request;
        System.out.println(operationCounter.countRequest(httpReq.getServletPath()));
    }

    @Override
    public void init(FilterConfig arg0) throws ServletException {
        operationCounter = new OperationCounter();
    }

}

File: tp-arithmetic-kotlin/src/main/kotlin/fr/isima/filters/CountTypeOfOperationFilter.kt
package fr.isima.filters

import fr.isima.service.OperationCounter
import javax.servlet.*
import javax.servlet.http.HttpServletRequest


/**
 * Compte les requêtes en fonction de leur type (math ou chaine)
 *
 * @author Benjamin Kuchcik
 */
class CountTypeOfOperationFilter(private val operationCounter: OperationCounter) : Filter {

    override fun doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain) {
        val httpReq = request as HttpServletRequest
        System.out.println(operationCounter.countRequest(httpReq.servletPath))
    }

    override fun destroy() {

    }

    override fun init(arg0: FilterConfig) {}

}
